### PR TITLE
bpo-23667: IDLE: Add trailing space highlight 

### PR DESCRIPTION
**Pull request by mlouielu**
https://bugs.python.org/issue23667

### CODE REVIEW

File: Lib/idlelib/colorizer.py [hunk 1]
```diff
@@ -20,15 +20,16 @@ def make_pat():
     # self.file = open("file") :
     # 1st 'file' colorized normal, 2nd as builtin, 3rd as string
     builtin = r"([^.'\"\\#]\b|^)" + any("BUILTIN", builtinlist) + r"\b"
+    trailing_space = any("TRAILINGSPACE", [r"[ \t]+(?=\n)"])
```
Comment (terryjreedy) [inline Lib/idlelib/colorizer.py:23]: I am not sure if these are the only whitespace chars to catch.  Does not matter for now.

File: Lib/idlelib/colorizer.py [hunk 2]
```diff
@@ -220,9 +226,17 @@ def recolorize_main(self):
                     for key, value in m.groupdict().items():
                         if value:
                             a, b = m.span(key)
-                            self.tag_add(key,
-                                         head + "+%dc" % a,
-                                         head + "+%dc" % b)
+                            if key == "TRAILINGSPACE" and b - a == 1:
+                                self.delay_render_trailing_space = self.after(
+                                    1000,
```
Comment (terryjreedy) [inline Lib/idlelib/colorizer.py:231]: Using colorizer avoids the problem of different sources.  The problem is that this is too short if the cursor is still on the line and too if it is leaving.

**Review by terryjreedy — COMMENTED** [review]
Marking whitespace as an error suggests that the user should fix it *now*, while it can be ignored until saving if strip on save is added.  But I will leave this open until the latter is added.

**Comment by rhettinger** [comment]
I'm thinking the strip-on-save option is the better way to go.

For my purposes, the visual indications of trailing whitespace would just be distracting noise that interferes with teaching.

**Comment by taleinat** [comment]
I agree that auto-stripping whitespace is the way to go, as long as it can be turned off in the configuration. See PR GH-17201.

I suggest closing this. @terryjreedy?

**Comment by terryjreedy** [comment]
@mlouielu Thank you for the patch.  We currently want to try the alternate solution.

**Pull request closed without merging**